odc/geo/cog/_mpu.py

```diff
@@ -240,46 +394,131 @@ def flush(
             bytes_written = self.flush_rhs(write)
 
         if self.left_data:
-            assert len(self.left_data) >= write.min_write_sz
-            partId = 1 if leftPartId is None else leftPartId
-            self.parts.insert(0, write(partId, self.left_data))
-            bytes_written += len(self.left_data)
-            self.left_data = bytearray()
-
+            if len(self.left_data) >= write.min_write_sz:
+                partId = self.nextPartId if leftPartId is None else leftPartId
+                self.parts.insert(0, write(partId, self.left_data))
+                bytes_written += len(self.left_data)
+                self.left_data = bytearray()
+            else:
+                logging.getLogger(__name__).debug(
+                    "Leftover data size (%s bytes) is less than min_write_sz; deferring flush.",
+                    len(self.left_data),
+                )
```

Comment:
NO: it's non-recoverable error and a sign or either misconfiguration or mistake in code